### PR TITLE
Do not use lookahead when parsing declare module or declare module.exports in flow

### PR DESCRIPTION
**Pull request by danez**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | 
| Patch: Bug Fix?          | n
| Major: Breaking Change?  | n
| Minor: New Feature?      | n
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

### CODE REVIEW

File: packages/babel-parser/src/plugins/flow.js [hunk 1]
```diff
@@ -422,8 +421,7 @@ export default (superClass: Class<Parser>): Class<Parser> =>
     flowParseDeclareModuleExports(
       node: N.FlowDeclareModuleExports,
     ): N.FlowDeclareModuleExports {
-      this.expectContextual("module");
-      this.expect(tt.dot);
```
Comment (xtuc) [inline packages/babel-parser/src/plugins/flow.js]: I might be wrong, but this was eating two tokens? Now it only skip one.

Comment (danez) [reply packages/babel-parser/src/plugins/flow.js]: right the other is skipped outside.

File: packages/babel-parser/src/plugins/flow.js [hunk 2]
```diff
@@ -240,12 +240,13 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       } else if (this.match(tt._var)) {
         return this.flowParseDeclareVariable(node);
       } else if (this.isContextual("module")) {
-        if (this.lookahead().type === tt.dot) {
+        this.next();
```
Comment (nicolo-ribaudo) [inline packages/babel-parser/src/plugins/flow.js]: Nit: you can use `this.eatContextual("module")` in the if condition

Comment (danez) [reply packages/babel-parser/src/plugins/flow.js]: done

**Pull request merged**